library/Network/CloudSeeder/Interfaces.hs

```diff
@@ -147,6 +158,9 @@ class Monad m => MonadCloud m where
   getStackInfo :: StackName -> m (Maybe Stack)
   getStackOutputs :: StackName -> m (Maybe (M.Map T.Text T.Text))
   runChangeSet :: T.Text -> m ()
+  encrypt :: T.Text -> T.Text -> m B.ByteString
+  upload :: T.Text -> T.Text -> B.ByteString -> m ()
+  generateSecret :: Int -> m T.Text
```

Comment:
I’m not sure if it makes complete sense for this to be a method of `MonadCloud`, since it doesn’t have anything to do with AWS, though it’s probably okay for now. Ideally, I would probably put it in its own typeclass (or even better, define it in terms of a minimal typeclass that only knows about randomness).